File: java/src/main/java/money_problem/domain/MissingExchangeRateException.java
package money_problem.domain;

public class MissingExchangeRateException extends Exception {
    public MissingExchangeRateException(Currency baseCurrency, Currency expectedCurrency) {
        super(String.format("Impossible to convert from %s currency to %s currency.", baseCurrency, expectedCurrency));
    }
}

File: php/src/MoneyProblem/Domain/MissingExchangeRateException.php
<?php

namespace MoneyProblem\Domain;

class MissingExchangeRateException extends \Exception
{

    /**
     * @param Currency $currency1
     * @param Currency $currency2
     */
    public function __construct(Currency $currency1, Currency $currency2)
    {
        parent::__construct(sprintf('%s->%s', $currency1, $currency2));

    }
}